service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -42,18 +48,93 @@ public DraftCMOService(DateFormatterService dateFormatterService, ObjectMapper m
         this.directionHelperService = directionHelperService;
     }
 
-    public DynamicList getHearingDateDynamicList(CaseDetails caseDetails) {
-        Map<String, Object> data = caseDetails.getData();
-        CaseData caseData = mapper.convertValue(data, CaseData.class);
+    public Map<String, Object> extractIndividualCaseManagementOrderObjects(
+        CaseManagementOrder caseManagementOrder,
+        List<Element<HearingBooking>> hearingDetails) {
 
-        List<Element<HearingBooking>> hearingDetails = caseData.getHearingDetails();
+        if (isNull(caseManagementOrder)) {
+            caseManagementOrder = CaseManagementOrder.builder().build();
+        }
+
+        HashMap<String, Object> data = new HashMap<>();
+        HashMap<String, Object> reviewCaseManagementOrder = new HashMap<>();
+
+        reviewCaseManagementOrder.put("cmoStatus", caseManagementOrder.getCmoStatus());
+
+        data.put("cmoHearingDateList", getHearingDateDynamicList(hearingDetails, caseManagementOrder));
+        data.put("schedule", caseManagementOrder.getSchedule());
+        data.put("recitals", caseManagementOrder.getRecitals());
+        data.put("reviewCaseManagementOrder", reviewCaseManagementOrder);
+
+        return data;
+    }
+
+    public CaseManagementOrder prepareCMO(Map<String, Object> caseData) {
+        DynamicList list = mapper.convertValue(caseData.get("cmoHearingDateList"), DynamicList.class);
+        Map<String, Object> reviewCaseManagementOrder = mapper.convertValue(
+            caseData.get("reviewCaseManagementOrder"), new TypeReference<>() {});
+        CMOStatus cmoStatus = mapper.convertValue(reviewCaseManagementOrder.get("cmoStatus"), CMOStatus.class);
+        Schedule schedule = mapper.convertValue(caseData.get("schedule"), Schedule.class);
+        List<Element<Recital>> recitals = mapper.convertValue(caseData.get("recitals"), new TypeReference<>() {});
+
+        return CaseManagementOrder.builder()
+            .hearingDate(list.getValue().getLabel())
+            .id(list.getValue().getCode())
+            .directions(combineAllDirectionsForCMO(mapper.convertValue(caseData, CaseData.class)))
+            .schedule(schedule)
+            .recitals(recitals)
+            .cmoStatus(cmoStatus)
+            .build();
+    }
+
+    public void prepareCaseDetails(Map<String, Object> caseData, CaseManagementOrder caseManagementOrder) {
+        final ImmutableSet<String> keysToRemove = ImmutableSet.of(
+            "cmoHearingDateList",
+            "schedule",
+            "reviewCaseManagementOrder",
+            "recitals");
+
+        keysToRemove.forEach(caseData::remove);
+
+        caseData.put("caseManagementOrder", caseManagementOrder);
+
+        switch (caseManagementOrder.getCmoStatus()) {
+            case SEND_TO_JUDGE:
+                // Does the same as PARTIES_REVIEW for now but in the future this will change
```

Comment:
I'd remove it, bit unnecessary. Code tells us that behaviour be the same as for next switch case. I do not care whether it will change later, it is how it is. We do not white roadmaps in the code.